src/core/auth/qgsauthmanager.cpp

```diff
@@ -398,6 +413,42 @@ const QString QgsAuthManager::disabledMessage() const
   return tr( "Authentication system is DISABLED:\n%1" ).arg( mAuthDisabledMessage );
 }
 
+bool QgsAuthManager::createAndStoreRandomMasterPasswordInKeyChain()
```

Comment:
There are many `return false` without error messages and the return value is not used when calling this function. IMHO, if this failed the user will not be able to understand why and what to do to fix the issue.

Reply:
There's already user visible errors here -- each of the methods here which return false and are possible to hit have their own error logging via message bar/message log. The remaining ones are just being extra safe and can't be hit (such as the redundant checks for isDisabled(), passwordHelperEnabled(), master password being empty -- none of those are possible given that they are also already checked before we attempt to generate the random password).